pkg/schedule/checker/checker_controller.go

```diff
@@ -352,25 +354,48 @@ func (c *Controller) CheckRegion(region *core.RegionInfo) []*operator.Operator {
 	}
 	// skip the joint checker, split checker and rule checker when region label is set to "schedule=deny".
 	// those checkers are help to make region health, it's necessary to skip them when region is set to deny.
-	if cl, ok := c.cluster.(interface{ GetRegionLabeler() *labeler.RegionLabeler }); ok {
-		l := cl.GetRegionLabeler()
-		if l.ScheduleDisabled(region) {
-			denyCheckersByLabelerCounter.Inc()
-			return nil
-		}
+	l := c.cluster.GetRegionLabeler()
+	if l.ScheduleDisabled(region) {
+		denyCheckersByLabelerCounter.Inc()
+		return nil
 	}
-	if c.mergeChecker != nil {
-		if ops := measureChecker(c.metrics.checkRegionHistograms[mergeChecker], func() []*operator.Operator {
-			if opController.OperatorCount(operator.OpMerge) < c.conf.GetMergeScheduleLimit() {
-				// It makes sure that two operators can be added successfully altogether.
-				return c.mergeChecker.Check(region)
+
+	if c.conf.IsAffinitySchedulingEnabled() {
+		if ops := measureChecker(c.metrics.checkRegionHistograms[affinityChecker], func() []*operator.Operator {
+			ops := c.affinityChecker.Check(region)
+			if len(ops) > 0 {
+				opKind := ops[0].Kind()
+				if (opKind & operator.OpMerge) == 0 {
```

Comment:
So there is no way to control the speed independently?

Reply:
It is not neccessary to introduce a new limit for affinity scheduling

Reply:
What if it affects the performance?

Reply:
I prefer to decouple these parameters.